client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java

```diff
@@ -217,10 +219,12 @@ private void requestExecutorMemory(long leastMem) {
       retry++;
       if (retry > requireMemoryRetryMax) {
         String message = "Can't get memory to cache shuffle data, request[" + askExecutorMemory
-            + "], got[" + gotMem + "]," + " WriteBufferManager allocated[" + allocatedBytes + "] task used[" + used
-            + "], consider to optimize 'spark.executor.memory'," + " 'spark.rss.writer.buffer.spill.size'.";
+            + "], got[" + gotMem + "]," + " WriteBufferManager allocated[" + allocatedBytes
+            + "] task used[" + used + "]. It may be caused by shuffle server is full of data"
+            + " or consider to optimize 'spark.executor.memory',"
+            + " 'spark.rss.writer.buffer.spill.size'.";
         LOG.error(message);
-        throw new OutOfMemoryError(message);
+        throw new RssCausedException(message);
```

Comment:
Should we make the exception classification more fine-grained, such as `RssCommitException`, `RssSendDataException`, etc.

Reply:
Yes, but current pr is target to throw RssCausedException and can be seen by spark directly, it will help event listener to catch it easily. The fine-grained exception can be created if necessary.

Reply:
OK